// @framer/benjamin.device/code/Device.tsx
import * as React from "react"
import { Frame, addPropertyControls, ControlType, Color } from "framer"

export function Device({
    children,
    image,
    color,
    mode,
    shadowBool,
    shadowAlpha,
    shadowDarken,
    radius,
    deviceWidth,
    deviceHeight,
    screenWidth,
    screenHeight,
}) {
    const currentColor = Color(color)
    let darkened = Color.darken(currentColor, shadowDarken)
    const dark = Color.toRgbString(Color.alpha(darkened, shadowAlpha))
    const mid = Color.toRgbString(Color.alpha(currentColor, 0.75))
    const lite = Color.toRgbString(Color.alpha(currentColor, 0.1))
    const frameColor = mode ? "#fff" : "#222"
    let innerRadius = radius / 2

    if (radius >= 50) {
        innerRadius = radius / 1.5
    }

    if (radius >= 75) {
        innerRadius = radius / 1.25
    }

    const cloned = children[0]
        ? React.cloneElement(children[0], {
              width: screenWidth,
              height: screenHeight,
          })
        : ""

    return (
        <>
            <Frame size={"100%"} backgroundColor={color} image={image}>
                <Frame
                    backgroundColor={frameColor}
                    center={true}
                    width={deviceWidth}
                    height={deviceHeight}
                    radius={radius}
                    style={{
                        boxShadow: mode
                            ? `inset 0 0 15px ${mid}, 0 20px 40px ${
                                  shadowBool ? dark : "transparent"
                              }`
                            : `inset 0 0 15px #000, 0 20px 40px ${
                                  shadowBool ? dark : "transparent"
                              }`,
                    }}
                    overflow={"hidden"}
                >
                    <Frame
                        width={screenWidth}
                        height={screenHeight}
                        center={true}
                        radius={innerRadius}
                        children={cloned}
                        background={lite}
                        overflow={"hidden"}
                    />
                </Frame>
            </Frame>
        </>
    )
}

Device.defaultProps = {
    width: 1920,
    height: 1080,
    color: "#ADF",
    image: "",
    mode: true,
    shadowBool: true,
    shadowAlpha: 0.5,
    shadowDarken: 30,
    radius: 60,
    deviceWidth: 413,
    deviceHeight: 850,
    screenWidth: 375,
    screenHeight: 812,
}

addPropertyControls(Device, {
    color: { type: ControlType.Color, title: "Color" },
    image: {
        type: ControlType.File,
        allowedFileTypes: ["jpg", "png", "jpeg"],
        title: "Image",
    },
    mode: {
        type: ControlType.Boolean,
        title: "Mode",
        enabledTitle: "Light",
        disabledTitle: "Dark",
    },
    radius: {
        type: ControlType.Number,
        title: "Radius",
        max: 100,
    },
    deviceWidth: {
        type: ControlType.Number,
        title: "Device W",
        min: 10,
        max: 3000,
    },
    deviceHeight: {
        type: ControlType.Number,
        title: "Device H",
        min: 10,
        max: 3000,
    },
    screenWidth: {
        type: ControlType.Number,
        title: "Screen W",
        min: 5,
        max: 3000,
    },
    screenHeight: {
        type: ControlType.Number,
        title: "Screen H",
        min: 5,
        max: 3000,
    },
    shadowBool: {
        type: ControlType.Boolean,
        title: "Shadow",
        enabledTitle: "On",
        disabledTitle: "Off",
    },
    shadowAlpha: {
        type: ControlType.Number,
        title: "↪ Alpha",
        min: 0,
        max: 1,
        step: 0.1,
    },
    shadowDarken: {
        type: ControlType.Number,
        title: "↪ Darken",
        min: 0,
        max: 100,
    },
})
