// @framer/marcopesani.font-awesome-free/code/FontAwesomeRegular.tsx
import * as React from "react";
import { PropertyControls, ControlType } from "framer";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import * as Icons from "@fortawesome/free-regular-svg-icons";

// Define type of property
interface Props {
  weight: string;
  icon: string;
  height: number;
  width: number;
  color: string;
  spin: boolean;
}

export class Regular extends React.Component<Props> {
  // Set default properties
  static defaultProps = {
    icon: "faAddressBook",
    height: 24,
    width: 24,
    color: "rgba(150,150,150,1)",
    spin: false
  };

  // Items shown in property panel
  static propertyControls: PropertyControls = {
    icon: {
      type: ControlType.Enum,
      title: "Icon",
      options: Object.keys(Icons[Icons.prefix]),
      optionTitles: Object.keys(Icons[Icons.prefix]).map(key => {
        const name = Icons[key].iconName.replace(/-/g, " ");
        return name.charAt(0).toUpperCase() + name.slice(1);
      })
    },
    color: { type: ControlType.Color, title: "Color" },
    spin: { type: ControlType.Boolean, title: "Spin" }
  };

  render() {
    return (
      <FontAwesomeIcon
        style={{ width: this.props.width, height: this.props.height }}
        color={this.props.color}
        icon={Icons[this.props.icon]}
        spin={this.props.spin}
        fixedWidth={true}
      />
    );
  }
}
