// @framer/marcopesani.font-awesome-free/code/FontAwesomeBrands.tsx
import * as React from "react";
import { PropertyControls, ControlType } from "framer";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import * as Icons from "@fortawesome/free-brands-svg-icons";

console.log(Icons)

// Define type of property
interface Props {
  height: number;
  width: number;
  icon: string;
  color: string;
  spin: boolean;
}

export class Brands extends React.Component<Props> {
  // Set default properties
  static defaultProps = {
    height: 24,
    width: 24,
    icon: "fa500px",
    color: "rgba(150,150,150,1)",
    spin: false
  };

  // Items shown in property panel
  static propertyControls: PropertyControls = {
    icon: {
      type: ControlType.Enum,
      title: "Icon",
      options: Object.keys(Icons[Icons.prefix])
    },
    color: { type: ControlType.Color, title: "Color" },
    spin: { type: ControlType.Boolean, title: "Spin" }
  };

  render() {
    return (
      <FontAwesomeIcon
        style={{ width: this.props.width, height: this.props.height }}
        color={this.props.color}
        icon={Icons[this.props.icon]}
        spin={this.props.spin}
        fixedWidth={true}
      />
    );
  }
}
