// @framer/bhaveshchow.credit-card/code/Card.tsx
import * as React from "react"
import Cards from "react-credit-cards"
import 'react-credit-cards/es/styles-compiled.css';
import {PropertyControls, ControlType } from 'framer'


interface Props{
    name: string
    number: string
    expiry: string
    cvc: string
    focused: "name" | "number" | "expiry" | "cvc"
    preview: boolean
    issuer: string
}

export class Card extends React.Component<Props> {
    static defaultProps = {
        height: 185,
        width: 290,
        name: "YOUR NAME HERE",
        focused: 'name',
        preview: false,
        number: '',
        expiry: '',
        cvc: '',
    }

    static propertyControls: PropertyControls = {
        name: {
            type: ControlType.String, title: "Name",
        },
        number: {
            type: ControlType.String, title: "Number",
        },
        expiry: {
            type: ControlType.String, title: "Expiry Date",
        },
        cvc: {
            type: ControlType.String, title: "CVC/CVV",
        },
        focused: {
            type: ControlType.Enum,
            title: "Focused",
            options: ["name", "number", "expiry", "cvc"],
        },
        preview: { type: ControlType.Boolean, title: "Preview" },
        issuer: {
            type: ControlType.Enum,
            title: "Issuer",
            options: ["amex", "dankort", "hipercard", "dinersclub", "discover", "jcb", "laser", "maestro", "mastercard", "unionpay", "visaelectron", "elo", "visa"],
            hidden(props) {
                return props.preview === false
            },
        },
    }

    render() {
        const {
            name,
            number,
            expiry,
            cvc,
            focused,
            preview,
            issuer,
        } = this.props
        const formattedNumber = number.replace(/\D+/g, '').slice(0, 19)
        const formattedExpiry = expiry.replace(/\D+/g, '').slice(0, 4)
        const formattedCvc = cvc.replace(/\D+/g, '').slice(0, 4)
        const componentProps = {
            name,
            formattedNumber,
            formattedExpiry,
            formattedCvc,
            focused,
            preview,
            issuer,
        }

        return (
            <div
                style={{
                    height: '100%',
                    display: 'flex',
                    alignItems: 'center',
                    justifyContent: 'center',
                    overflow:'hidden',
                    textAlign:'center',
                }}
            >
                
                <Cards
                    {...componentProps}
                    number={formattedNumber}
                    expiry={formattedExpiry}
                    cvc={formattedCvc}
                />
            </div>
        )
    }
}
