// @framer/hjotar.browser-x-kit/code/BrowserBottom.tsx
import * as React from "react";
import { PropertyControls, ControlType } from "framer";

interface Props {
    browser: string;
    homeBar: string;
};

export class BrowserBottom extends React.Component<Props> {

    static defaultProps = {
        browser: "safari",
        height: 82,
        homeBar: "show",
        width: 375,
    }

    static propertyControls: PropertyControls = {
        browser: {
            type: ControlType.Enum,
            options: ["safari", "chrome"],
            optionTitles: ["Safari", "Chrome"],
            title: "Browser"
        },
        homeBar: {
            type: ControlType.SegmentedEnum,
            options: ["show", "hide"],
            optionTitles: ["Show", "Hide"],
            title: "Home Bar"
        },
    }

    render() {
        const { homeBar, browser } = this.props;

        return (
            <div
                style={{
                    background: browser === "safari" ? "#F9F9F9" : "#F8F8F8",
                    boxShadow: browser === "safari" ? "0px -0.5px #B2B2B2" : "none",
                    color: "#414141",
                    display: "flex",
                    fontWeight: "normal",
                    fontSize: 17,
                    height: 82,
                    overflow: "hidden",
                    position: "relative",
                    width: "100%",
                }}
            >
                <div 
                    style={{
                        alignItems: "baseline",
                        background: "transparent",
                        display: "flex",
                        height: "38",
                        justifyContent: "space-between",
                        marginTop: browser === "safari" ? 11 : 5,
                        marginLeft: browser === "safari" ? 9 : 13,
                        marginRight: browser === "safari" ? 16 : 13,
                        position: "relative",
                        width: "100%",
                    }} 
                >

                    <div
                        dangerouslySetInnerHTML={{ 
                            __html: browser === "safari" ? aleft_safari() : aleft_chrome()
                        }}
                    />
                    
                    <div
                        style={{ 
                            left: browser === "safari" ? 3 : 4,
                            position: "relative",
                        }} 
                        dangerouslySetInnerHTML={{ 
                            __html: browser === "safari" ? aright_safari() : aright_chrome()
                        }}
                    />

                    <div 
                        style={{
                            background: "#E1E1E1", 
                            borderRadius: 19,
                            display: browser === "safari" ? "none" : "block",
                            height: 38,
                            width: 70,
                        }}
                    >
                        <div
                            style={{ 
                                margin: "auto",
                                marginTop: 7,
                                paddingRight: 3,
                                textAlign: "center",
                            }}
                            dangerouslySetInnerHTML={{ 
                                __html: search_chrome() 
                            }}
                        />
                    </div>

                    <div
                        style={{ 
                            left: 3,
                            display: browser === "safari" ? "block" : "none",
                            position: "relative",
                        }} 
                        dangerouslySetInnerHTML={{ 
                            __html: share_safari()
                        }}
                    />

                    <div 
                        style={{ 
                            background: "transparent",
                            right: browser === "safari" ? -2 : 5,
                            position: "relative",
                        }} 
                    >
                        <div
                            style={{ 
                                margin: "auto",
                                left: 0,
                                right: 0,
                                position: "relative",
                                textAlign: "center",
                            }}
                            dangerouslySetInnerHTML={{
                                __html: browser === "safari" ? bookmark_safari() : page_chrome()
                            }}
                        />
                        <div 
                            style={{
                                color: "#7B7B7B",
                                display: browser === "safari" ? "none" : "block",
                                fontFamily: "SF-Pro-Text, system-ui, -apple-system",
                                fontWeight: "bold",
                                fontSize: 13,
                                left: 10,
                                position: "absolute",
                                top: 4,
                            }} 
                        >
                        1
                        </div>
                    </div>

                    <div
                        dangerouslySetInnerHTML={{ 
                            __html: browser === "safari" ? page_safari() : more_chrome()
                        }}
                    />

                </div>

                <div
                    style={{
                        background: "#000000",
                        borderRadius: 100,
                        bottom: 8,
                        display: homeBar === "show" ? "block" : "none",
                        height: 5,
                        left:0,
                        margin: "0 auto",
                        position: "absolute",
                        right: 0,
                        width: 133,
                    }}
                />
                
            </div>
        );
    }
}

function aleft_chrome() {
    return `<?xml version="1.0" encoding="UTF-8"?>
    <svg width="26px" height="24px" viewBox="0 0 26 24" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
        <g id="aleft" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd" stroke-linecap="round" stroke-linejoin="round">
            <polyline id="aleft" stroke="#7B7B7B" stroke-width="3" fill-rule="nonzero" points="18 2 8.5 12 18 22"></polyline>
        </g>
    </svg>`
}

function aright_chrome() {
    return `<?xml version="1.0" encoding="UTF-8"?>
    <svg width="26px" height="24px" viewBox="0 0 26 24" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
        <g id="aright" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd" stroke-linecap="round" stroke-linejoin="round">
            <polyline id="aright" stroke="#C5C6C7" stroke-width="3" fill-rule="nonzero" transform="translate(12.750000, 12.000000) scale(-1, 1) translate(-12.750000, -12.000000) " points="17.5 2 8 12 17.5 22"></polyline>
        </g>
    </svg>`
}

function search_chrome() {
    return `<?xml version="1.0" encoding="UTF-8"?>
    <svg width="26px" height="24px" viewBox="0 0 26 24" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
        <g id="search" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
            <path d="M19.2093358,15.4630889 C19.2344127,15.4845487 19.2588511,15.5071448 19.2825835,15.5308772 L24.5890649,20.8373586 C25.0847273,21.3330211 25.0847273,22.1366487 24.5890649,22.6323112 C24.0934024,23.1279737 23.2897747,23.1279737 22.7941122,22.6323112 L17.4876308,17.3258298 C17.4506778,17.2888768 17.4164798,17.2502122 17.3850367,17.210091 C15.7812922,18.3860557 13.7958165,19.0815942 11.6460962,19.0815942 C6.31870439,19.0815942 2,14.8100338 2,9.5407971 C2,4.27156036 6.31870439,0 11.6460962,0 C16.9734881,0 21.2921925,4.27156036 21.2921925,9.5407971 C21.2921925,11.7783669 20.5134188,13.83603 19.2093358,15.4630889 Z M11.6460962,16.5708581 C15.5715429,16.5708581 18.7537461,13.4233926 18.7537461,9.5407971 C18.7537461,5.65820161 15.5715429,2.51073608 11.6460962,2.51073608 C7.72064961,2.51073608 4.53844638,5.65820161 4.53844638,9.5407971 C4.53844638,13.4233926 7.72064961,16.5708581 11.6460962,16.5708581 Z" fill="#707070" fill-rule="nonzero"></path>
        </g>
    </svg>`
}

function page_chrome() {
    return `<?xml version="1.0" encoding="UTF-8"?>
    <svg width="26px" height="24px" viewBox="0 0 26 24" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
        <g id="page" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
            <path d="M1,6 C1,2.686 3.686,0 7,0 L19,0 C22.314,0 25,2.686 25,6 L25,18 C25,21.314 22.314,24 19,24 L7,24 C3.686,24 1,21.314 1,18 L1,6 Z M3.5,17.7 C3.5,19.79855 5.20145,21.5 7.3,21.5 L18.7,21.5 C20.79855,21.5 22.5,19.79855 22.5,17.7 L22.5,6.3 C22.5,4.20145 20.79855,2.5 18.7,2.5 L7.3,2.5 C5.20145,2.5 3.5,4.20145 3.5,6.3 L3.5,17.7 Z" fill="#7B7B7B" fill-rule="nonzero"></path>
        </g>
    </svg>`
}

function more_chrome() {
    return `<?xml version="1.0" encoding="UTF-8"?>
    <svg width="26px" height="24px" viewBox="0 0 26 24" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
        <g id="more" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
            <path d="M3,15 C1.34314575,15 0,13.6568542 0,12 C0,10.3431458 1.34314575,9 3,9 C4.65685425,9 6,10.3431458 6,12 C6,13.6568542 4.65685425,15 3,15 Z M13,15 C11.3431458,15 10,13.6568542 10,12 C10,10.3431458 11.3431458,9 13,9 C14.6568542,9 16,10.3431458 16,12 C16,13.6568542 14.6568542,15 13,15 Z M23,15 C21.3431458,15 20,13.6568542 20,12 C20,10.3431458 21.3431458,9 23,9 C24.6568542,9 26,10.3431458 26,12 C26,13.6568542 24.6568542,15 23,15 Z" fill="#7B7B7B" fill-rule="nonzero"></path>
        </g>
    </svg>`
}

function aleft_safari() {
    return `<?xml version="1.0" encoding="UTF-8"?>
    <svg width="26px" height="30px" viewBox="0 0 26 30" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
        <g id="aleft_safari" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
            <path d="M8.00320068,16.014987 L8,16.0117399 L9.10728371,14.8877078 L9.11078731,14.8912624 L16.8887225,7 L18,8.1270295 L10.2218542,16.0185057 L17.9670511,23.8764675 L16.8597674,25 L9.1142984,17.1421989 L9.11127752,17.1452638 L8,16.0182343 L8.00320068,16.014987 Z" id="aleft" fill="#0078FF"></path>
        </g>
    </svg>`
}

function aright_safari() {
    return `<?xml version="1.0" encoding="UTF-8"?>
    <svg width="26px" height="30px" viewBox="0 0 26 30" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
        <g id="aright_safari" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
            <path d="M8.00320068,16.014987 L8,16.0117399 L9.10728371,14.8877078 L9.11078731,14.8912624 L16.8887225,7 L18,8.1270295 L10.2218542,16.0185057 L17.9670511,23.8764675 L16.8597674,25 L9.1142984,17.1421989 L9.11127752,17.1452638 L8,16.0182343 L8.00320068,16.014987 Z" id="aright" fill="#DDDDDD" transform="translate(13.000000, 16.000000) scale(-1, 1) translate(-13.000000, -16.000000) "></path>
        </g>
    </svg>`
}

function share_safari() {
    return `<?xml version="1.0" encoding="UTF-8"?>
    <svg width="26px" height="30px" viewBox="0 0 26 30" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
        <g id="share_safari" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
            <path d="M12.4931854,2.914 L9.34784424,6.0605 L8.64110462,5.3535 L12.9930013,1 L17.344898,5.3535 L16.6381583,6.0605 L13.4928171,2.914 L13.4928171,18.207 L12.4931854,18.207 L12.4931854,2.914 Z M22.4965,27.707 L3.4965,27.707 L3.4965,8.707 L10.493922,8.707 L10.493922,9.707 L4.49613171,9.707 L4.49613171,26.707 L21.4968683,26.707 L21.4968683,9.707 L15.4920806,9.707 L15.4920806,8.707 L22.4965,8.707 L22.4965,27.707 Z" id="share" fill="#0078FF"></path>
        </g>
    </svg>`
}

function bookmark_safari() {
    return `<?xml version="1.0" encoding="UTF-8"?>
    <svg width="26px" height="30px" viewBox="0 0 26 30" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
        <g id="bookmark_safari" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
            <path d="M19.156,4 C16.5275,4 14.333,5.1065 13,7.0635 C11.6675,5.1065 9.4725,4 6.844,4 C3.78,4 0.5,5.6395 0.5,8.08 L0.5,26.423 C0.5,26.616 0.605,26.793 0.774,26.8845 C0.9435,26.975 1.1495,26.966 1.308,26.86 C1.3485,26.833 5.3935,24.164 8.6555,24.164 C10.769,24.164 12.0225,25.2795 12.4895,27.5755 C12.4925,27.591 12.502,27.604 12.5065,27.619 C12.515,27.6475 12.5255,27.674 12.5385,27.7005 C12.5535,27.7305 12.57,27.7575 12.59,27.7835 C12.6065,27.8055 12.624,27.826 12.644,27.845 C12.668,27.8685 12.694,27.888 12.7225,27.906 C12.745,27.9205 12.767,27.935 12.7925,27.9455 C12.825,27.96 12.8585,27.968 12.894,27.976 C12.9125,27.98 12.9285,27.99 12.948,27.992 C12.965,27.9935 12.982,27.9945 12.9995,27.9945 L12.9995,27.9945 L13,27.9945 L13.0005,27.9945 L13.001,27.9945 C13.018,27.9945 13.035,27.993 13.0525,27.992 C13.0715,27.99 13.0875,27.9805 13.106,27.976 C13.141,27.968 13.175,27.9605 13.2075,27.9455 C13.2325,27.9345 13.2545,27.9205 13.2775,27.906 C13.306,27.888 13.332,27.8685 13.356,27.845 C13.376,27.826 13.393,27.8055 13.4095,27.7835 C13.4295,27.7575 13.446,27.7305 13.4605,27.7005 C13.4735,27.674 13.484,27.6475 13.4925,27.619 C13.497,27.604 13.5065,27.591 13.5095,27.5755 C13.9765,25.2795 15.2305,24.164 17.3435,24.164 C20.606,24.164 24.6505,26.8325 24.691,26.86 C24.85,26.9655 25.0555,26.975 25.225,26.8845 C25.394,26.793 25.499,26.616 25.499,26.423 L25.499,8.08 C25.5,5.6395 22.2205,4 19.156,4 L19.156,4 Z M8.656,23.1165 C5.972,23.1165 2.956,24.6655 1.542,25.4885 L1.542,8.08 C1.542,6.4935 4.0695,5.0485 6.8445,5.0485 C9.3545,5.0485 11.401,6.1985 12.4795,8.2095 L12.4795,24.886 C11.6195,23.72 10.3295,23.1165 8.656,23.1165 L8.656,23.1165 Z M24.4585,25.4885 C23.0445,24.6655 20.0285,23.1165 17.3445,23.1165 C15.671,23.1165 14.381,23.72 13.521,24.886 L13.521,8.21 C14.5995,6.199 16.6465,5.049 19.156,5.049 C21.931,5.049 24.458,6.494 24.458,8.0805 L24.458,25.4885 L24.4585,25.4885 Z" id="bookmark" fill="#0078FF"></path>
        </g>
    </svg>`
}

function page_safari() {
    return `<?xml version="1.0" encoding="UTF-8"?>
    <svg width="26px" height="30px" viewBox="0 0 26 30" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
        <g id="page_safari" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
            <path d="M0.75,8.5 L19.75,8.5 L19.75,27.5 L0.75,27.5 L0.75,8.5 Z M1.75,9.5 L1.75,26.5 L18.75,26.5 L18.75,9.5 L1.75,9.5 Z M20.75,22 L20.75,21 L24.25,21 L24.25,4 L7.25,4 L7.25,7.5 L6.25,7.5 L6.25,3 L25.25,3 L25.25,22 L20.75,22 Z" id="Combined-Shape" fill="#0078FF" fill-rule="nonzero"></path>
        </g>
    </svg>`
}
