// @framer/hjotar.browser-x-kit/code/BrowserTop.tsx
import * as React from "react";
import { PropertyControls, ControlType } from "framer";
import { string } from "prop-types";

interface Props {
    browser: string;
    statusBar: string;
    text: string;
};

export class BrowserTop extends React.Component<Props> {

    static defaultProps = {
        browser: "safari",
        height: 94,
        statusBar: "show",
        text: "yourwebsite.com",
        width: 375,
    }
    
    static propertyControls: PropertyControls = {
        browser: {
            type: ControlType.Enum,
            options: ["safari", "chrome"],
            optionTitles: ["Safari", "Chrome"],
            title: "Browser"
        },
        statusBar: {
            type: ControlType.SegmentedEnum,
            options: ["show", "hide"],
            optionTitles: ["Show", "Hide"],
            title: "Status Bar"
        },
        text: {
            type: ControlType.String, title: "URL"
        }
    }
    
    render() {
        const { statusBar, browser } = this.props;
        return (
            <div
                style={{
                    background: browser === "safari" ? "#F9F9F9" : "#F4F4F5",
                    boxShadow: browser === "safari" ? "0px 0.5px #B2B2B2" : "none",
                    color: "#414141",
                    display: "flex",
                    fontFamily: "SF-Pro-Text, system-ui, -apple-system",
                    fontWeight: browser === "safari" ? "500" : "normal",
                    fontSize: 17,
                    height: browser === "safari" ? 94 : 92,
                    overflow: "hidden",
                    position: "relative",
                    width: "100%",
                }}
            >
                <div
                    style={{
                        display: statusBar === "show" ? "block" : "none",
                    }}
                >
                
                    <div
                        style={{
                            alignItems: "center",
                            color: "#000000",
                            display:"flex",
                            fontSize: "15px",
                            fontFamily: "SF-Pro-Text, system-ui, -apple-system",
                            fontWeight: "600",
                            height: 20,
                            justifyContent: "center",
                            left: 14,
                            lineHeight: "20px",
                            padding: 0,
                            position: "absolute",
                            top: 12,
                            width: 68,
                        }}
                    >
                        <div>
                            9:41
                        </div>
                        <div
                            style={{
                                height: 20
                            }}
                            dangerouslySetInnerHTML={{
                                __html: location()
                            }}
                        />
                    </div>

                    <div
                    style={{ 
                        display:"flex",
                        padding: 0,
                        position: "absolute",
                        right: 14,
                        top: 12,
                    }}
                >
                    <div
                        dangerouslySetInnerHTML={{ 
                            __html: signal() 
                        }}
                    />
                    <div 
                        style={{ 
                            marginLeft: 5
                        }}
                        dangerouslySetInnerHTML={{ 
                            __html: wifi() 
                        }}
                    />
                    <div 
                        style={{ 
                            marginLeft: 5 
                        }}
                        dangerouslySetInnerHTML={{ 
                            __html: battery() 
                        }}
                    />
                </div>
                
                </div>
                <div 
                    style={{
                        alignItems: "center",
                        background: browser === "safari" ? "#E7E8E9" : "#DBDBDC",
                        borderRadius: browser === "safari" ? 10 : 18,
                        bottom: browser === "safari" ? 10 : 6,
                        color: browser === "safari" ? "#000000" : "#414141",
                        fontFamily: browser === "safari" ? "HelveticaNeue, system-ui, -apple-system" : "SF-Pro-Text, system-ui, -apple-system",
                        display: "flex",
                        height: 36,
                        justifyContent: "center",
                        left: browser === "safari" ? 10 : 8,
                        position: "absolute",
                        right: browser === "safari" ? 10 : 8,
                    }}
                >
                    <div
                        style={{ 
                            float: "left",
                            marginLeft: 0,
                            marginRight: browser === "safari" ? 11 : 9,
                            marginTop: browser === "safari" ? 0 : 1,
                        }}
                        dangerouslySetInnerHTML={{ 
                            __html: browser === "safari" ? lock_safari() : lock_chrome()
                        }}
                    />
                    <div>
                        {this.props.text}
                    </div>
                    <div
                        style={{
                            position: "absolute",
                            right: browser === "safari" ? 8 : 14,
                            top: browser === "safari" ? 8 : 7,
                        }}
                        dangerouslySetInnerHTML={{
                            __html: browser === "safari" ? refresh() : share()
                        }}
                    />
                </div>
            </div>
        );
    }
}

function location() {
    return `<?xml version="1.0" encoding="UTF-8"?>
    <svg width="16px" height="20px" viewBox="0 0 16 20" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
        <g id="Location" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd" stroke-linejoin="round">
            <g id="location" transform="translate(3.000000, 6.000000)" stroke="#000000">
                <polygon id="Shape" points="5.86818182 9.54545455 10 0 0.454545455 4.13181818 5.86818182 4.13181818"></polygon>
            </g>
        </g>
    </svg>`
}

function signal() {
    return `<?xml version="1.0" encoding="UTF-8"?>
    <svg width="17.5px" height="12px" viewBox="0 0 18 12" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
        <g id="Signal" stroke="none" fill="none" fill-rule="evenodd">
            <g id="signal" transform="translate(-293.000000, -17.000000)" fill-rule="nonzero" fill="#000000">
            <path d="M295.666667,24.3333333 C296.218951,24.3333333 296.666667,24.7810486 296.666667,25.3333333 L296.666667,27.3333333 C296.666667,27.8856181 296.218951,28.3333333 295.666667,28.3333333 L294.666667,28.3333333 C294.114382,28.3333333 293.666667,27.8856181 293.666667,27.3333333 L293.666667,25.3333333 C293.666667,24.7810486 294.114382,24.3333333 294.666667,24.3333333 L295.666667,24.3333333 Z" opacity="1"></path>
            <path d="M299.333333,22.3333333 L300.333333,22.3333333 C300.885618,22.3333333 301.333333,22.7810486 301.333333,23.3333333 L301.333333,27.3333333 C301.333333,27.8856181 300.885618,28.3333333 300.333333,28.3333333 L299.333333,28.3333333 C298.781049,28.3333333 298.333333,27.8856181 298.333333,27.3333333 L298.333333,23.3333333 C298.333333,22.7810486 298.781049,22.3333333 299.333333,22.3333333 Z" opacity="1"></path>
            <path d="M304,20 L305,20 C305.552285,20 306,20.4477153 306,21 L306,27.3333333 C306,27.8856181 305.552285,28.3333333 305,28.3333333 L304,28.3333333 C303.447715,28.3333333 303,27.8856181 303,27.3333333 L303,21 C303,20.4477153 303.447715,20 304,20 Z" opacity="1"></path>
            <path d="M308.666667,17.6666667 L309.666667,17.6666667 C310.218951,17.6666667 310.666667,18.1143819 310.666667,18.6666667 L310.666667,27.3333333 C310.666667,27.8856181 310.218951,28.3333333 309.666667,28.3333333 L308.666667,28.3333333 C308.114382,28.3333333 307.666667,27.8856181 307.666667,27.3333333 L307.666667,18.6666667 C307.666667,18.1143819 308.114382,17.6666667 308.666667,17.6666667 Z" opacity="1"></path>
            </g>
        </g>
    </svg>`
}

function wifi() {
    return `<?xml version="1.0" encoding="UTF-8"?>
    <svg width="15px" viewBox="0 0 16 12" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
        <g id="Wifi" fill="none" fill-rule="evenodd">
            <g id="wifi" transform="translate(-315.000000, -17.000000)" fill="#000000" fill-rule="nonzero">
            <path d="M325.549333,26.188 C325.611087,26.127394 325.645095,26.0439919 325.643328,25.957485 C325.641561,25.8709782 325.604176,25.7890342 325.54,25.731 C324.264428,24.6521162 322.396238,24.6521162 321.120667,25.731 C321.056446,25.7889875 321.019,25.8709057 321.01717,25.957413 C321.01534,26.0439203 321.049289,26.1273487 321.111,26.188 L323.108667,28.2036667 C323.167217,28.2629057 323.247042,28.2962443 323.330333,28.2962443 C323.413624,28.2962443 323.49345,28.2629057 323.552,28.2036667 L325.549333,26.188 Z" opacity="1"></path>
            <path d="M323.330333,23.4036667 C324.547823,23.4035921 325.721865,23.8561245 326.624333,24.6733333 C326.746397,24.7893139 326.938678,24.7867994 327.057667,24.6676667 L328.212333,23.501 C328.273141,23.439805 328.30688,23.3567887 328.306003,23.2705239 C328.305126,23.1842592 328.269706,23.101946 328.207667,23.042 C325.459468,20.4856177 321.203532,20.4856177 318.455333,23.042 C318.393256,23.1019453 318.357837,23.1842997 318.357023,23.2705923 C318.356208,23.356885 318.390065,23.4398934 318.451,23.501 L319.605333,24.6676667 C319.724322,24.7867994 319.916603,24.7893139 320.038667,24.6733333 C320.940539,23.8566645 322.113649,23.4041708 323.330333,23.4036667 Z" opacity="1"></path>
            <path d="M323.330333,19.608 C325.546226,19.6080971 327.677386,20.4595347 329.283333,21.9863333 C329.404265,22.1042079 329.597562,22.102721 329.716667,21.983 L330.872667,20.8163333 C330.932975,20.7556112 330.966599,20.6733595 330.966098,20.5877792 C330.965598,20.5021988 330.931014,20.420346 330.87,20.3603333 C326.654902,16.3207858 320.005098,16.3207858 315.79,20.3603333 C315.728941,20.4203015 315.694297,20.5021294 315.693734,20.5877099 C315.693171,20.6732903 315.726736,20.755567 315.787,20.8163333 L316.943333,21.983 C317.062362,22.1029017 317.255808,22.1043897 317.376667,21.9863333 C318.982819,20.4594344 321.114223,19.6079944 323.330333,19.608 Z" opacity="1"></path>
            </g>
        </g>
    </svg>`
}

function battery() {
    return `<?xml version="1.0" encoding="UTF-8"?>
    <svg width="25px" height="12px" viewBox="0 0 25 12" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
        <g id="Battery" stroke="none" fill="none" fill-rule="evenodd">
            <g id="battery" transform="translate(-336.000000, -17.000000)">
                <g id="battery" transform="translate(336.000000, 17.000000)">
                    <rect id="Border" stroke="#000000" opacity="0.35" x="0.5" y="0.833333333" width="21" height="10.3333333" rx="2.66666675"></rect>
                    <path d="M23,4 L23,8 C23.8047311,7.66122348 24.328038,6.87313328 24.328038,6 C24.328038,5.12686672 23.8047311,4.33877652 23,4" id="Cap" fill="#000000" fill-rule="nonzero" opacity="0.4"></path>
                    <rect id="Capacity" fill="#000000" fill-rule="nonzero" x="2" y="2.33333333" height="7.33333333" rx="1.33333337" width="18"></rect>
                </g>
            </g>
        </g>
    </svg>`
}

function lock_chrome() {
    return `<?xml version="1.0" encoding="UTF-8"?>
    <svg width="10px" height="13px" viewBox="0 0 20 26" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
        <g id="Lock" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
            <path d="M4,8 L4,6 C4,2.6862915 6.6862915,6.08718376e-16 10,0 C13.3137085,-6.08718376e-16 16,2.6862915 16,6 L16,8 L18,8 C19.1045695,8 20,8.8954305 20,10 L20,24 C20,25.1045695 19.1045695,26 18,26 L2,26 C0.8954305,26 1.3527075e-16,25.1045695 0,24 L0,10 C-1.3527075e-16,8.8954305 0.8954305,8 2,8 L4,8 Z M10,2 C7.790861,2 6,3.790861 6,6 L6,8 L14,8 L14,6 C14,3.790861 12.209139,2 10,2 Z" id="Combined-Shape" fill="#424242" fill-rule="nonzero"></path>
        </g>
    </svg>`
}

function share() {
    return `<?xml version="1.0" encoding="UTF-8"?>
    <svg width="16px" height="20px" viewBox="0 0 16 20" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
        <g id="Share" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
            <path d="M9,3.41421356 L9,12 C9,12.5522847 8.55228475,13 8,13 C7.44771525,13 7,12.5522847 7,12 L7,3.40685425 L5.87132034,4.53553391 C5.48079605,4.9260582 4.84763107,4.9260582 4.45710678,4.53553391 C4.06658249,4.14500961 4.06658249,3.51184464 4.45710678,3.12132034 L7.28553391,0.292893219 C7.48172834,0.0966987867 7.73916357,-0.000930060595 7.99630641,6.67685163e-06 L8,0 C8.27614237,0 8.52614237,0.111928813 8.70710678,0.292893219 L11.5355339,3.12132034 C11.9260582,3.51184464 11.9260582,4.14500961 11.5355339,4.53553391 C11.1450096,4.9260582 10.5118446,4.9260582 10.1213203,4.53553391 L9,3.41421356 Z M4.5,8 L3.5,8 C2.67157288,8 2,8.67157288 2,9.5 L2,16.5 C2,17.3284271 2.67157288,18 3.5,18 L12.5,18 C13.3284271,18 14,17.3284271 14,16.5 L14,9.5 C14,8.67157288 13.3284271,8 12.5,8 L10.5,8 L10.5,6 L13,6 C14.6568542,6 16,7.34314575 16,9 L16,17 C16,18.6568542 14.6568542,20 13,20 L3,20 C1.34314575,20 0,18.6568542 0,17 L0,9 C0,7.34314575 1.34314575,6 3,6 L4.5,6 L4.5,8 Z" id="Combined-Shape" fill="#424242" fill-rule="nonzero"></path>
        </g>
    </svg>`
}

function refresh() {
    return `<?xml version="1.0" encoding="UTF-8"?>
    <svg width="16px" height="20px" viewBox="0 0 16 20" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
        <g id="refresh" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
            <path d="M8.00664509,3.80093773 L8.00664509,1.5 L13.4371316,4.96873917 L8.00664509,8.43805617 L8.00664509,5.53444105 C4.81765697,5.53725041 2.23350283,8.12371418 2.23350283,11.3127817 C2.23350283,14.5041604 4.82046689,17.0917023 8.0118456,17.0917023 C10.9963596,17.0917023 13.4527332,14.7919219 13.7584075,11.8888825 C14.3639778,11.8883046 15.0654686,11.8963943 15.5,11.8963943 C15.19317,15.7638391 11.9578759,18.8246273 8.0118456,18.8246273 C3.86299549,18.8246273 0.5,15.4616318 0.5,11.3127817 C0.5,7.16624282 3.86018555,3.80374724 8.00664509,3.80093785 Z" id="Refresh" fill="#000000"></path>
        </g>
    </svg>`
}

function lock_safari() {
    return `<?xml version="1.0" encoding="UTF-8"?>
    <svg width="10px" height="14px" viewBox="0 0 10 14" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
        <g id="lock_safari" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
            <path d="M1.5,6 L1.5,3.5 C1.5,1.84314575 2.84314575,0.5 4.5,0.5 L5.5,0.5 C7.15685425,0.5 8.5,1.84314575 8.5,3.5 L8.5,6 L9,6 C9.55228475,6 10,6.44771525 10,7 L10,13 C10,13.5522847 9.55228475,14 9,14 L1,14 C0.44771525,14 6.76353751e-17,13.5522847 0,13 L0,7 C-6.76353751e-17,6.44771525 0.44771525,6 1,6 L1.5,6 Z M7.5,6 L7.5,3.5 C7.5,2.3954305 6.6045695,1.5 5.5,1.5 L4.5,1.5 C3.3954305,1.5 2.5,2.3954305 2.5,3.5 L2.5,6 L7.5,6 Z" id="Combined-Shape" fill="#000000" fill-rule="nonzero"></path>
        </g>
    </svg>`
}
